### ISSUE TITLE
implement logging best practices

### ISSUE THREAD

**Issue by pohly** [kubernetes-sigs/dra-example-driver#10]
Currently the example driver uses logrus for logging. Logrus pioneered structured logging, but nowadays has been superseded by other, more modern solutions. Let's use something else.

**Comment by pohly** [comment]
My proposal is to use the same logging technology as in Kubernetes:
- The main binary calls k8s.io/component-base/logs to set up logging - see https://github.com/kubernetes/component-base/tree/master/logs/example.
- Let's enable contextual logging by default, as in https://github.com/kubernetes-csi/external-provisioner/pull/639
- Everything else uses [contextual logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/3077-contextual-logging).

**Comment by byako** [comment]
+1 for contextual logging

**Comment by pohly** [comment]
The code already uses component-base/logs...

**Comment by pohly** [comment]
... but not in all binaries.

**Comment by pohly** [comment]
Since f6acdbae37, all commands have these flags:
```
   Logging:

   --feature-gates value  A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:
     AllAlpha=true|false (ALPHA - default=false)
     AllBeta=true|false (BETA - default=false)
     ContextualLogging=true|false (ALPHA - default=false)
     LoggingAlphaOptions=true|false (ALPHA - default=false)
     LoggingBetaOptions=true|false (BETA - default=true) (default: ContextualLogging=true) [$FEATURE_GATES]
   --log-flush-frequency value        Maximum number of seconds between log flushes (default: 5s) [$LOG_FLUSH_FREQUENCY]
   --log-json-info-buffer-size value  [Alpha] In JSON format with split output streams, the info messages can be buffered for a while to increase performance. The default value of zero bytes disables buffering. The size can be specified as number of bytes (512), multiples of 1000 (1K), multiples of 1024 (2Ki), or powers of those (3M, 4G, 5Mi, 6Gi). Enable the LoggingAlphaOptions feature gate to use this. (default: 0) [$LOG_JSON_INFO_BUFFER_SIZE]
   --log-json-split-stream value      [Alpha] In JSON format, write error messages to stderr and info messages to stdout. The default is to write a single stream to stdout. Enable the LoggingAlphaOptions feature gate to use this. (default: false) [$LOG_JSON_SPLIT_STREAM]
   --logging-format value             Sets the log format. Permitted formats: "json" (gated by LoggingBetaOptions), "text". (default: text) [$LOGGING_FORMAT]
   --vmodule value                    comma-separated list of pattern=N settings for file-filtered logging (only works for text log format) [$VMODULE]
   -v value                           number for the log level verbosity (default: 0) [$V]
```

In other words, they support JSON, text output and contextual logging. `ContextualLogging=true` is the default feature gate flag, which enables contextual logging.

**Comment by pohly** [comment]
What was missing are some log call conversions and automated testing. https://github.com/kubernetes-sigs/dra-example-driver/pull/39 fixes this.
